src/pages/VideoDetails/VideoDetails.js

```diff
@@ -1,6 +1,333 @@
+                  >
+                    <div
+                      className={`flex items-center flex-1 justify-between ${styles.statItem}`}
+                    >
+                      <span className={`material-icons-round ${styles.icon}`}>
+                        visibility
+                      </span>
+                      <div className={`space-x-0-5`}></div>
+                      <span>{video.viewCount} views</span>
+                    </div>
+                    <div
+                      className={`flex items-center flex-1 justify-between ${styles.statItem}`}
+                    >
+                      <span className={`material-icons-round ${styles.icon}`}>
+                        favorite
+                      </span>
+                      <div className={`space-x-0-5`}></div>
+                      <span>{video.likeCount} likes</span>
+                    </div>
+                  </div>
+                  <div
+                    className={`flex justify-around py-2 ${styles.stats} w-full`}
+                  >
+                    <div
+                      className={`flex items-center flex-1 justify-between ${styles.statItem}`}
+                    >
+                      <span className={`material-icons-round ${styles.icon}`}>
+                        forum
+                      </span>
+                      <div className={`space-x-0-5`}></div>
+                      <span>{video.commentCount} comments</span>
+                    </div>
+                    <div
+                      className={`flex items-center flex-1 justify-between ${styles.statItem}`}
+                    >
+                      <span className={`material-icons-round ${styles.icon}`}>
+                        timelapse
+                      </span>
+                      <div className={`space-x-0-5`}></div>
+                      <span>{video.publishedDate}</span>
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      )}
+    </div>
+  ) : (
+    <Loading />
+  );
 }
+
+const AuthModal = ({ setShowAuthModal }) => {
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
+
+  return (
+    <div className={`${styles.modalOuter}`}>
+      <div className={`${styles.modalInner}`}>
+        <h2>Uh Oh!</h2>
+        <p>You need to login in order to add videos to your playlists.</p>
+        <div className="flex items-center justify-center">
+          <button
+            onClick={() => setShowAuthModal(false)}
+            className="btn btn-outlined btn-small"
+          >
+            Cancel
+          </button>
+          <button
+            onClick={() => navigate("/login", { state: { from: pathname } })}
+            className="btn btn-solid btn-small"
+          >
+            Login
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+};
```

Comment:
Try to split file into pieces, if the code exceeds 200 Lines. 

You can - 
- Extract the Auth Modal as Component. 
- Split the code into different Components, as it may seem fit.
- Extract the functions into a different file, and Import here, to reduce the file size.